solution-jivanov/models/users.js

```diff
@@ -0,0 +1,283 @@
+// users.js
+// users model class
+
+const uuidv4 = require('uuid/v4');
+const validate = require('validate.js');
+
+const Model = require('../lib/model.js');
+
+const ID_VALIDATION_CONSTRAINTS = {
+    presence: true, // it is mandatory
+    type: 'string', // ID must be of type string, since it is an UUID
+    length: {
+        is: 36, // and must be exactly 36 chars too
+        message: 'The record ID must be exactly 36 chars long'
+    },
+};
+const CREATE_VALIDATION_CONSTRAINTS = {
+    name: {
+        presence: true, // only name is mandatory
+        length: {
+            minimum: 3, // and must be at least 3 chars long
+            message: 'field is mandatory and must be at least 3 chars long'
+        },
+    },
+    dob: {
+        presence: false,
+    },
+    address: {
+        presence: false,
+        length: {
+            minimum: 2, // and must be at least 2 chars long
+            message: 'must be at least 2 chars long'
+        },
+    },
+    description: {
+        presence: false,
+        length: {
+            minimum: 2, // and must be at least 2 chars long
+            message: 'must be at least 2 chars long'
+        },
+    },
+};
+const UPDATE_VALIDATION_CONSTRAINTS = {
+    name: {
+        presence: false, // only name is mandatory
+        length: {
+            minimum: 3, // and must be at least 3 chars long
+            message: 'must be at least 3 chars long'
+        },
+    },
+    dob: {
+        presence: false,
+    },
+    address: {
+        presence: false,
+        length: {
+            minimum: 2, // and must be at least 2 chars long
+            message: 'must be at least 2 chars long'
+        },
+    },
+    description: {
+        presence: false,
+        length: {
+            minimum: 2, // and must be at least 2 chars long
+            message: 'must be at least 2 chars long'
+        },
+    },
+};
+const VALIDATION_OPTIONS = {
+    format: 'flat', // return "flat" message format i.e. a single string
+};
+
+const USERS_TABLE_NAME = process.env.USERS_TABLE_NAME;

```

Comment:
Would you see any value in using destructuring syntax here?

Reply:
In this particular case...mmm...don't see anything obvious really...process.env is always an object...and yes, with destructuring you could set a default, but you could do that without it too...at the same time this is easier to read than destructuring...please let me know if I am missing anything...